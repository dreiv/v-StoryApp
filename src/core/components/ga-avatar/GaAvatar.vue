<script setup lang="ts">
export interface GaAvatarProps {
  size?: 'small' | 'medium' | 'large'
  content?: string
}

const { size = 'medium', content } = defineProps<GaAvatarProps>()
</script>

<template>
  <div :class="[$style.avatar, $style[size]]">
    {{ content }}
  </div>
</template>

<style module>
.avatar {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  border-radius: var(--ga-radius-round);
  background-color: var(--ga-color-surface-action);
  color: var(--ga-color-text-on-action);
  font-weight: var(--ga-font-weight-medium);
  font-family: var(--ga-font-family-primary);
}

.small {
  width: var(--ga-size-spacing-06);
  height: var(--ga-size-spacing-06);
  font-size: var(--ga-text-sm-font-size);
}

.medium {
  width: var(--ga-size-spacing-08);
  height: var(--ga-size-spacing-08);
  font-size: var(--ga-text-md-font-size);
}

.large {
  width: var(--ga-size-spacing-10);
  height: var(--ga-size-spacing-10);
  font-size: var(--ga-text-lg-font-size);
}
</style>
