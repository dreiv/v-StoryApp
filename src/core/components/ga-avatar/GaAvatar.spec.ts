import { shallowMount, type VueWrapper } from '@vue/test-utils'
import GaAvatar from './GaAvatar.vue'

describe('GaAvatar', () => {
  let wrapper: VueWrapper

  it('should render correctly', () => {
    wrapper = shallowMount(GaAvatar)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.element.tagName).toBe('DIV')
  })

  it('should render with content', () => {
    wrapper = shallowMount(GaAvatar, { props: { content: 'AV' } })
    expect(wrapper.text()).toBe('AV')
  })

  it('should apply size classes when size prop is provided', async () => {
    wrapper = shallowMount(GaAvatar)
    expect(wrapper.classes()).toContain('medium')

    await wrapper.setProps({ size: 'small' })
    expect(wrapper.classes()).toContain('small')

    await wrapper.setProps({ size: 'large' })
    expect(wrapper.classes()).toContain('large')
  })

  it('should render as a button when interactive prop is true', async () => {
    wrapper = shallowMount(GaAvatar, { props: { interactive: true } })
    expect(wrapper.element.tagName).toBe('BUTTON')
    expect(wrapper.classes()).toContain('interactive')
  })

  it('should render an image when image prop is provided', () => {
    const imgSrc = 'https://example.com/avatar.jpg'
    wrapper = shallowMount(GaAvatar, { props: { image: imgSrc } })

    const img = wrapper.find('img')
    expect(img.exists()).toBe(true)
    expect(img.attributes('src')).toBe(imgSrc)
  })

  it('should fall back to content when image fails to load', async () => {
    wrapper = shallowMount(GaAvatar, {
      props: {
        image: 'invalid-url.jpg',
        content: 'AV',
      },
    })

    // Trigger image error
    await wrapper.find('img').trigger('error')

    // Should show the fallback content
    expect(wrapper.text()).toBe('AV')
  })
})
