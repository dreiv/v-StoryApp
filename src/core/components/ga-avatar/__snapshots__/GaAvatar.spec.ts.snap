// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GaAvatar > should render correctly 1`] = `
"<div class="avatar medium">
  <!--v-if-->
</div>"
`;
